### PR TITLE
feature-T1-130-warn-user-about-taken-phone: introduced warning modal

### PR DESCRIPTION
**Pull request by andreyzhylin**

### CODE REVIEW

File: src/Lykke.Service.OAuth/wwwroot/js/app/components/signup/accountInfo/accountInfo.controller.js [hunk 1]
```diff
@@ -35,10 +35,16 @@
...
                 }).catch(function (error) {
-                    var invalidPhoneFormatError = 'InvalidPhoneFormat';
+                    var errorCodes = {
+                        phoneNumberInUse: 'PhoneNumberInUse',
+                        invalidPhoneFormatError: 'InvalidPhoneFormat'
+                    };
                     vm.data.isSubmitting = false;
-                    vm.data.isPhoneFormatInvalid = invalidPhoneFormatError === error.data.error;
-                    // TODO: Password exists
+                    vm.data.isPhoneFormatInvalid = errorCodes.invalidPhoneFormatError === error.data.error;
```
Comment (gmaf) [inline src/Lykke.Service.OAuth/wwwroot/js/app/components/signup/accountInfo/accountInfo.controller.js]: Use switch here.

File: src/Lykke.Service.OAuth/wwwroot/js/app/components/signup/accountInfo/accountInfo.controller.js [hunk 2]
```diff
@@ -35,10 +35,16 @@
                     vm.data.isSubmitting = false;
                     $scope.$emit('currentStepChanged', signupStep.pin);
                 }).catch(function (error) {
-                    var invalidPhoneFormatError = 'InvalidPhoneFormat';
+                    var errorCodes = {
```
Comment (gmaf) [inline src/Lykke.Service.OAuth/wwwroot/js/app/components/signup/accountInfo/accountInfo.controller.js]: Move to global error constants to separate section ```PhoneErrorCodes```.

File: src/Lykke.Service.OAuth/wwwroot/js/app/components/signup/accountInfo/accountInfo.controller.js (hunk 1 above)
Comment (andreyzhylin) [reply src/Lykke.Service.OAuth/wwwroot/js/app/components/signup/accountInfo/accountInfo.controller.js]: Fixed

File: src/Lykke.Service.OAuth/wwwroot/js/app/components/signup/accountInfo/accountInfo.controller.js (hunk 2 above)
Comment (andreyzhylin) [reply src/Lykke.Service.OAuth/wwwroot/js/app/components/signup/accountInfo/accountInfo.controller.js]: Fixed

**Pull request merged**